ui/raidboss/popup-text.ts

```diff
@@ -978,113 +1199,174 @@ export class PopupText {
     // infoText
-    const lowerTextKey = textType + 'Text';
+    const lowerTextKey = textMap[textType].text;
     // InfoText
-    const upperTextKey = textTypeUpper + 'Text';
-    const textObj = triggerHelper.triggerOptions[upperTextKey] ||
-      triggerHelper.trigger[lowerTextKey] || triggerHelper.response[lowerTextKey];
+    const upperTextKey = textMap[textType].upperText;
+    // InfoSound
+    const upperSound = textMap[textType].upperSound;
+    // InfoSoundVolume
+    const upperSoundVolume = textMap[textType].upperSoundVolume;
+
+    let textObj: RaidbossTriggerOutput =
+      triggerHelper.triggerOptions[upperTextKey];
+    if (!textObj && triggerHelper.trigger[lowerTextKey])
+      textObj = triggerHelper.trigger[lowerTextKey] as RaidbossTriggerOutput;
+    if (!textObj && triggerHelper.response)
+      textObj = triggerHelper.response[lowerTextKey];
     if (textObj) {
-      const text = triggerHelper.valueOrFunction(textObj);
-      triggerHelper.defaultTTSText = triggerHelper.defaultTTSText || text;
-      if (text && triggerHelper.textAlertsEnabled) {
+      let text = triggerHelper.valueOrFunction(textObj);
+      if (!text)
+        return;
+      if (typeof text === 'number')
+        text = text.toString();
+      if (typeof text !== 'string')
+        text = String(text);
+      triggerHelper.defaultTTSText = triggerHelper.defaultTTSText ?? text;
+      if (text && typeof text === 'string' && triggerHelper.textAlertsEnabled) {
         // per-trigger option > trigger field > option duration by text type
-        const duration = triggerHelper.duration.fromConfig ||
-          triggerHelper.duration.fromTrigger || triggerHelper.duration[lowerTextKey];
+        let duration = triggerHelper.duration?.fromConfig ||
+          triggerHelper.duration?.fromTrigger;
+        if (duration === undefined && triggerHelper.duration)
+          duration = triggerHelper.duration[lowerTextKey];
+        if (duration === undefined)
+          duration = 0;
+
         this._createTextFor(triggerHelper, text, textType, lowerTextKey, duration);
         if (!triggerHelper.soundUrl) {
-          triggerHelper.soundUrl = this.options[textTypeUpper + 'Sound'];
-          triggerHelper.soundVol = this.options[textTypeUpper + 'SoundVolume'];
+          triggerHelper.soundUrl = this.options[upperSound];
+          triggerHelper.soundVol = this.options[upperSoundVolume];
         }
       }
     }
   }
 
-  _makeTextElement(triggerHelper, text, className) {
+  _makeTextElement(_triggerHelper: TriggerHelper, text: string, className: string): HTMLElement {
     const div = document.createElement('div');
     div.classList.add(className);
     div.classList.add('animate-text');
     div.innerText = text;
     return div;
   }
 
-  _playAudioFile(triggerHelper, url, volume) {
+  _playAudioFile(triggerHelper: TriggerHelper, url: string, volume: number): void {
     const audio = new Audio(url);
     audio.volume = volume || 1;
```

Comment:
```suggestion
    audio.volume = volume ?? 1;
```

Reply:
I adjusted some other stuff around this, there was a difference in logic I accidentally introduced, so please re-review this one.

Reply:
Looks good, thanks for pointing it out.